internal/server/apimigrator.go

```diff
@@ -26,23 +27,22 @@ type apiMigration struct {
 	redirect        string
 	requestRewrite  func(c *gin.Context)
 	responseRewrite func(c *gin.Context)
+	redirectHandler func(c *gin.Context)
+	index           int
 }
 
-func addRedirect(a *API, method, path, newPath, version string) {
-	// update any existing migrations with the legacy path
-	for i, mig := range a.migrations {
-		if len(mig.redirect) == 0 && mig.path == path {
-			a.migrations[i].path = newPath
-		}
-		if len(mig.redirect) > 0 && mig.redirect == path {
-			a.migrations[i].redirect = newPath
-		}
+func addRedirect(a *API, method, path, newPath, version string, optMiddleware ...gin.HandlerFunc) {
+	var optRedirectMiddleware gin.HandlerFunc
+	if len(optMiddleware) > 0 {
+		optRedirectMiddleware = optMiddleware[0]
```

Comment:
Setting the redirect middleware to the first optRedirectMiddleware means it will go through the whole chain of middleware if there are many?

Reply:
this just allows addRedirect to have a single optional custom middleware handler. There should only ever be one, but most redirects don't have it. There's an example of this in the tests:
```go
	addRedirect(a, "get", "/oldtest/:id", "/awesometest", "0.1.3", func(c *gin.Context) {
		id := c.Param("id")
		q := c.Request.URL.Query()
		q.Add("vegetableCount", id)
		c.Request.URL.RawQuery = q.Encode()
		c.Next()
	})
```